File: 1-reverse-number/ruan-brandao/reverse_number.ex
defmodule ReverseNumber do
  @moduledoc """
  Module containing a function to reverse an integer.
  """

  @doc """
  Reverts an integer

  ## Examples
      iex> ReverseNumber.reverse(12345)
      54321
  """
  def reverse(number) when is_integer(number) do
    number
    |> Integer.digits()
    |> Enum.reverse()
    |> Enum.join()
    |> String.to_integer()
  end

  def reverse(_number), do: {:error, "Invalid Input"}
end

File: 1-reverse-number/VitorLuizC/reverseNumber.ts
const reverse = (
  value: bigint,
  reversion: bigint
): bigint => (
  value === 0n 
    ? reversion
    : reverse(value / 10n, reversion * 10n + value % 10n)   
);

const reverseNumber = (
  value: number | bigint
): number => Number(reverse(BigInt(value), 0n));

export default reverseNumber;
